src/choreo/tower/fd_tower.c

```diff
@@ -201,12 +201,12 @@ fd_tower_switch_check( fd_tower_t const * tower,
 }
 
 int
-fd_tower_threshold_check( fd_tower_t const *    tower,
-                          fd_epoch_t const *    epoch,
-                          fd_funk_t *           funk,
-                          fd_funk_txn_t const * txn,
-                          ulong                 slot,
-                          fd_tower_t *          scratch ) {
+fd_tower_threshold_check( fd_tower_t const *                 tower,
+                          fd_epoch_t *                       epoch,
+                          fd_replay_out_vote_state_t const * vote_states,
+                          ulong                              vote_states_len,
```

Comment:
this should pass in an array of `fd_tower_t` instead  so `fd_tower.h` doesn't need to know about the calling layer (replay_out and tiles) -- i didn't particularly like the old way of passing in `funk` and `txn` but it did it for performance / for the "fork" model